Synthetic tokens explainer (#1039)
UMAprotocol/protocol: documentation/explainers/synthetic_tokens.md

## Launching a New Synthetic Token
To launch a new type of synthetic token for which an existing market does not yet exist, that synthetic token’s smart contract must first be parameterized and deployed. Anyone can parameterize and deploy this contract. 

This smart contract can be deployed using UMA’s synthetic token contract template and defining key parameters. Some parameters to highlight are:
- token’s price identifier (the price feed this token should track)
- token expiration timestamp
- token collateralization ratio (e.g. a synthetic token must have collateral worth at least 120% of the price indentifier’s current value) 

This [tutorial](../tutorials/creating_synthetic_tokens_from_truffle_console.md) will show you how to parameterize and deploy the smart contract for a new synthetic token from the command line.

## Creating Synthetic Tokens From an Existing Contract
After the contract is deployed, anyone can interact with the contract to create synthetic tokens. These actors are called “token sponsors”, since they sponsor the creation of synthetic tokens. 

<p align="center">
  <img src="st_create_contract.png" width="500">
</p>

> Token sponsors deposit collateral into the contract to collateralize synthetic tokens, which they can then withdraw and trade with others. The first token sponsor to create synthetic tokens is able to immediately withdraw synthetic tokens from the contract. Any following token sponsors who wish to create synthetic tokens can do so, but must collateralize their positions by at least as much as the system’s global collateralization ratio. This prevents token sponsor positions from being immediately liquidated. <!-- TODO: add link to glossary for “global collateralization ratio” --> 

Reviewer: "This prevents token sponsor positions from being immediately liquidated"... Allison pointed out to me that this isn't technically correct. They can still dispute (there's no rule against that), they would just be wrong. Can we rephrase?

Author: I agree that this isn't technically correct, especially because it doesn't describe the mechanism by which it is effected, but I believe it conveys the point that we are trying to make - that collateralizing to the GCR provides some assurance that your position will not be immediately liquidated. 

What do you think of adding this paragraph as a separate footnote that we can link to? 

The global collateralization ratio (GCR) is the average collateralization ratio among all token sponsor positions that have not yet been liquidated nor have any outstanding withdrawal requests. Requiring new token sponsors to collateralize their positions by as much as the GCR provides some assurances that so long as those token sponsors below the GCR have not yet been liquidated, those above the GCR need not be at risk of liquidation.

Reviewer: I think your paragraph is great and should live here (in addition to having an (even more detailed?) version in the glossary. I don't think it should be a footnote.

Can you add this paragraph and rework modestly to make it flow? I think we should also change the ""This prevents token sponsor positions from being immediately liquidated" to something like "The insures that all newly created tokens sponsor positions are correctly collateralized upon initialization".

Author: Yes, I'll do this in the next commit.

> Token sponsors deposit collateral into the contract to collateralize synthetic tokens, which they can then withdraw and trade with others. The first token sponsor to create synthetic tokens is able to immediately withdraw synthetic tokens from the contract. Any following token sponsors who wish to create synthetic tokens can do so, but must collateralize their positions by at least as much as the system’s global collateralization ratio. This prevents token sponsor positions from being immediately liquidated. <!-- TODO: add link to glossary for “global collateralization ratio” --> 

Author: Sure, and in fact, that was the intention behind adding the sentence, "This prevents token sponsor positions from being immediately liquidated." Is there another way you'd prefer to express why the GCR matters? 

So far, all links to this glossary only exist in "TODO" comments. We can prioritize creating a glossary next so that we have something to link to here, if you don't want us to have a lot of comments asking for a glossary? Or we can remove the comments and instead create a Github issue?

> Token sponsors deposit collateral into the contract to collateralize synthetic tokens, which they can then withdraw and trade with others. The first token sponsor to create synthetic tokens is able to immediately withdraw synthetic tokens from the contract. Any following token sponsors who wish to create synthetic tokens can do so, but must collateralize their positions by at least as much as the system’s global collateralization ratio. This prevents token sponsor positions from being immediately liquidated. <!-- TODO: add link to glossary for “global collateralization ratio” --> 

Reviewer: ```suggestion
Token sponsors deposit collateral into the contract to collateralize synthetic tokens, which they can then withdraw and trade with others. The first token sponsor to create synthetic tokens is able to immediately withdraw synthetic tokens from the contract. Any following token sponsors who wish to create synthetic tokens can do so, but must collateralize their positions by at least as much as the system’s global collateralization ratio. This prevents token sponsor positions from being immediately liquidated. 

The global collateralization ratio (GCR) is the average collateralization ratio among all token sponsor positions that have not yet been liquidated nor have any outstanding withdrawal requests. Requiring new token sponsors to collateralize their positions by as much as the GCR provides some assurances that so long as those token sponsors below the GCR have not yet been liquidated, those above the GCR need not be at risk of liquidation.
<!-- TODO: add link to glossary for “global collateralization ratio” --> 
```

> Token sponsors deposit collateral into the contract to collateralize synthetic tokens, which they can then withdraw and trade with others. The first token sponsor to create synthetic tokens is able to immediately withdraw synthetic tokens from the contract. Any following token sponsors who wish to create synthetic tokens can do so, but must collateralize their positions by at least as much as the system’s global collateralization ratio. This prevents token sponsor positions from being immediately liquidated. <!-- TODO: add link to glossary for “global collateralization ratio” --> 

Reviewer: ```suggestion
Token sponsors deposit collateral into the contract to collateralize synthetic tokens, which they can then withdraw and trade with others. The first token sponsor to create synthetic tokens is able to immediately withdraw synthetic tokens from the contract. Any following token sponsors who wish to create synthetic tokens can do so, but must collateralize their positions by at least as much as the system’s global collateralization ratio. The insures that all newly created tokens sponsor positions are correctly collateralized upon initialization.

The global collateralization ratio (GCR) is the average collateralization ratio among all token sponsor positions that have not yet been liquidated nor have any outstanding withdrawal requests. Requiring new token sponsors to collateralize their positions by as much as the GCR provides some assurances that so long as those token sponsors below the GCR have not yet been liquidated, those above the GCR need not be at risk of liquidation. <!-- TODO: add link to glossary for “global collateralization ratio” --> 
```

> This tutorial will show you how to mint tokens from the command line after having parameterized and deployed the synthetic token’s smart contract. <!-- TODO: add link -->

Author: Right, here is where I needed your/engineering help on attaching a link not only to another file (which I can do), but a link to an anchor in another file. Let me know if we should add this as a blocker or get other engineers involved.

> This tutorial will show you how to mint tokens from the command line after having parameterized and deployed the synthetic token’s smart contract. <!-- TODO: add link -->

Reviewer: ```suggestion
This [tutorial](../tutorials/creating_synthetic_tokens_from_truffle_console.md#create-new-tokens-from-an-existing-contract) will show you how to mint tokens from the command line after having parameterized and deployed the synthetic token’s smart contract. <!-- TODO: add link -->
```

> This tutorial will show you how to mint tokens from the command line after having parameterized and deployed the synthetic token’s smart contract. <!-- TODO: add link -->

Reviewer: @rcai1 I believe that link above will work when clicking the link from the GitHub docs directory. Aka when that suggestion is accepted clicking from https://github.com/UMAprotocol/protocol/blob/rcai1-patch-1/documentation/explainers/synthetic_tokens.md should take you to https://github.com/UMAprotocol/protocol/blob/master/documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md#create-new-tokens-from-an-existing-contract

HOWEVER, I do not think this link will compile correctly when viewed from docs.umaproject.org. We'll need @mrice32 (or engineering) to fix this. Let's just worry about getting the GitHub right for now and ask Matt/team to help with the rest.

> This tutorial will show you how to mint tokens from the command line after having parameterized and deployed the synthetic token’s smart contract. <!-- TODO: add link -->

Reviewer: Matt added a GitHub issues for this

## Managing Token Sponsor Positions
During the lifetime of the synthetic token, token sponsors may want to deposit additional collateral to their positions so as to avoid liquidation. Sponsors may also want to withdraw excess collateral if the position has moved in their favor. 

Token sponsors can deposit additional collateral at any time.

<p align="center">
  <img src="st_add_collateral.png" align="center" width="500">
</p>

Token sponsors can withdraw excess collateral in one of two ways: a “fast” withdrawal or “slow” withdrawal.

### “Fast” withdrawal:

> A “fast” withdrawal allows a token sponsor to withdraw excess collateral from his position immediately, so long as the resulting position is collateralized by at least as much as the average token sponsor collateralization ratio for the contract (“global collateralization ratio”). <!-- TODO: add link to glossary for “global collateralization ratio” --> 

Author: I've adapted a part of the GCR explainer paragraph above to add to the "Fast" withdrawal section.

<p align="center">

>   <img src="st_withdraw_collateral.png" align="center" width="500">

Author: Yes, we should move this to the next section.

> </p>

Reviewer: ```suggestion
```

In a “slow” withdrawal, there are two parts: 
The token sponsor submits a withdrawal request to the contract indicating the amount of collateral he wishes to withdraw and the timestamp of the request.

During this period, any token holder can liquidate the token sponsor’s position if they believe a withdrawal of the amount indicated in the withdrawal request would bring the token sponsor’s collateralization below the “collateralization requirement” at the time of liquidation. <!-- TODO: add link to glossary for “collateralization requirement” --> 
If the “withdrawal liveness period” passes without a token holder liquidating the token sponsor, the token sponsor may withdraw collateral from his position up to the amount requested. <!-- TODO: add link to glossary for “withdrawal liveness period” --> 

> 

Reviewer: ```suggestion
<p align="center">
  <img src="st_withdraw_collateral.png" align="center" width="500">
</p>

```

# Overview

> Synthetic tokens are collateral-backed tokens whose required backing collateral fluctuates depending on the tokens’ reference index. Synthetic tokens blend features of prediction markets, futures markets, and collateralized loans. 

Reviewer: ```suggestion
Synthetic tokens are collateral-backed tokens whose value fluctuates depending on the tokens’ reference index. Synthetic tokens blend features of prediction markets, futures markets, and collateralized loans. 
```

For more details on how a “priceless” framework can be applied more generally to other types of financial contract templates on the blockchain, check out this blog post. 

## Launching a New Synthetic Token

To launch a new type of synthetic token for which an existing market does not yet exist, that synthetic token’s smart contract must first be parameterized and deployed. Anyone can parameterize and deploy this contract. 

This smart contract can be deployed using UMA’s synthetic token contract template and defining key parameters. Some parameters to highlight are:
- token’s price identifier (the price feed this token should track)
- token expiration timestamp
- token collateralization ratio (e.g. a synthetic token must have collateral worth at least 120% of the price indentifier’s current value) 

This tutorial will show you how to parameterize and deploy the smart contract for a new synthetic token from the command line.

## Creating Synthetic Tokens From an Existing Contract

> After the contract is deployed, anyone can interact with the contract to create synthetic tokens. These actors are called “token sponsors”, since they sponsor the creation of synthetic tokens. [Image]

Reviewer: ```suggestion
After the contract is deployed, anyone can interact with the contract to create synthetic tokens. These actors are called “token sponsors”, since they sponsor the creation of synthetic tokens. 
![Diagram](./synthetic_token_creation.png)
```

In a “slow” withdrawal, there are two parts: 
The token sponsor submits a withdrawal request to the contract indicating the amount of collateral he wishes to withdraw and the timestamp of the request.

During this period, any token holder can liquidate the token sponsor’s position if they believe a withdrawal of the amount indicated in the withdrawal request would bring the token sponsor’s collateralization below the “collateralization requirement” at the time of liquidation. [“collateralization requirement” is a link to a glossary of terms.] 
If the “withdrawal liveness period” passes without a token holder liquidating the token sponsor, the token sponsor may withdraw collateral from his position up to the amount requested. [“withdrawal liveness period” is a link to a glossary of terms.] 

## Liquidation and Dispute

> At any time, a token holder may liquidate a token sponsor’s position. Liquidations happen immediately without calling the oracle. Anyone may dispute a liquidation within the “liquidation liveness period”. [“withdrawal liveness period” is a link to a glossary of terms.] 

Author: Because they are two separate parameters in the parameterization of a contract for a new synthetic token, I would call them "withdrawal liveness period" and "liquidation liveness period".

To liquidate a token sponsor position, a token holder submits tokens to the contract. The tokens are submitted for 3 purposes: to indicate the size of the position to be liquidated, to close the token sponsor’s position, and to post a bond attesting to the liquidator’s belief that the token sponsor’s position should be liquidated. The liquidator will lose a portion of this bond if their liquidation is disputed and found to be invalid.

Here are three ways in which a liquidation can be resolved: 

> 1. No one disputes the liquidation during the liquidation liveness period. The token sponsor’s position is liquidated after the liquidation liveness period ends. All collateral deposited by the token sponsor is returned to the liquidator. 

Author: We can remove the second sentence and replace it with the third. I think the third sentence is accurate, although it seems like you are suggesting we change it to highlight the difference between partial and complete liquidations? If so, we can include a numerical example to make the proportional return of collateral clear.

1. Someone disputes the liquidation during the liquidation liveness period. To do this, the disputer must post a bond. Once the dispute is raised, a request is made of the UMA DVM to resolve the dispute and indicate whether the token sponsor was undercollateralized at the time of the liquidation. [“DVM” is a link to a glossary of terms.] 
- If the DVM decides that the token sponsor was undercollateralized at the time of the liquidation:
    * The disputer will lose their bond. 
    * The liquidator will receive all of the token sponsor position’s collateral. 
    * The token sponsor will not receive any of the collateral they have previously deposited into the position. 
- If the DVM decides that the token sponsor was not undercollateralized at the time of the liquidation: 
    * The disputer will receive back their dispute bond and a dispute reward.

>     * The liquidator will receive collateral equal to the price identifier at that time, less a dispute reward.

Reviewer: ```suggestion
    * The liquidator will receive collateral equalling: (i) the value of the token at the time of liquidation as determined by the DVM, less (ii) the dispute reward paid to the disputer, less (iii) the improper liquidation reward paid to the original token sponsor.
```

# Overview

Synthetic tokens are collateral-backed tokens whose value fluctuates depending on the tokens’ reference index. Synthetic tokens blend features of prediction markets, futures markets, and collateralized loans. 

Some examples of synthetic tokens include: 
- synthetic real-world assets
- synthetic cross-chain cryptoassets
- tracking tokens for various non-tradable indices

> Some of the most creative ideas for synthetic tokens fall in the last category. Check out our [Slack channel #tokenideas](https://join.slack.com/t/umaprotocol/shared_invite/zt-7mtxxds5-OIhE~q_WkwGCVNrq0~G~rg) for more project ideas like:

Reviewer 2: ```suggestion
Some of the most creative ideas for synthetic tokens fall in the last category. Check out our [Slack channel #tokenideas](https://join.slack.com/t/umaprotocol/shared_invite/enQtNTk4MjQ4ODY0MDA1LTM4ODg0NGZhYWZkNjkzMDE4MjU0ZGFlYWQzZTFiZWFlZjI2NDE4OGI2NWY3OTdhYjYyZjg0MjAzMTgwODVhZTE) for more project ideas like:
```

    * The disputer will receive back their dispute bond and a dispute reward.
    * The liquidator will receive collateral equalling: (i) the value of the token at the time of liquidation as determined by the DVM, less (ii) the dispute reward paid to the disputer, less (iii) the improper liquidation reward paid to the original token sponsor.
    * The token sponsor will receive any remaining collateral and a reward for the improper liquidation. 

## Redeeming Tokens

Before the expiration date of the token, tokens may only be redeemed by token sponsors. A token sponsor redeems a token by submitting it to the contract to be burned and receiving collateral proportional to the total amount of collateral that the token sponsor has deposited to the contract. 

Consider the following example. Assume a token sponsor has deposited 150 DAI of collateral to create 100 synthetic tokens, which they then sold to the market. Later, the token sponsor repurchases 30 tokens and decides to redeem them. These 30 tokens represent 30% of the token sponsors original position, so by redeeming them the token sponsor receives 30% of their initial collateral, or 45 DAI (since 30%*150 = 45).

> ### Redeeming After Expiry

Reviewer 2: Are my eyes deceiving me...? Is this text and below italicized? Is that on purpose?

Author: Not on purpose! Must have something to do with the % sign...

# Overview

Synthetic tokens are collateral-backed tokens whose value fluctuates depending on the tokens’ reference index. Synthetic tokens blend features of prediction markets, futures markets, and collateralized loans. 

Some examples of synthetic tokens include: 

> - synthetic real-world assets

Reviewer 3: do you want to give an example here in brackets?
```suggestion
- synthetic real-world assets (eg: gold or Tesla stock price)
```

Author: I think it may be better to leave them without examples for now (or potentially link to NFP/things like US Stocks) in the future.

In a “slow” withdrawal, there are two parts: 
The token sponsor submits a withdrawal request to the contract indicating the amount of collateral he wishes to withdraw and the timestamp of the request.

During this period, any token holder can liquidate the token sponsor’s position if they believe a withdrawal of the amount indicated in the withdrawal request would bring the token sponsor’s collateralization below the “collateralization requirement” at the time of liquidation. <!-- TODO: add link to glossary for “collateralization requirement” --> 
If the “withdrawal liveness period” passes without a token holder liquidating the token sponsor, the token sponsor may withdraw collateral from his position up to the amount requested. <!-- TODO: add link to glossary for “withdrawal liveness period” --> 

## Liquidation and Dispute

> At any time, a token holder may liquidate a token sponsor’s position. Liquidations happen immediately without calling the oracle. Anyone may dispute a liquidation within the “liquidation liveness period”. <!-- TODO: add link to glossary for “withdrawal liveness period” --> 

Reviewer 3: nit: I think you meant Liquidation liveness period here.
```suggestion
At any time, a token holder may liquidate a token sponsor’s position. Liquidations happen immediately without calling the oracle. Anyone may dispute a liquidation within the “liquidation liveness period”. <!-- TODO: add link to glossary for “Liquidation liveness period” --> 
```
